store/src/java/com/zimbra/cs/index/event/Event.java

```diff
@@ -0,0 +1,73 @@
+package com.zimbra.cs.index.event;
+
+import java.util.Map;
+
+public class Event {
+    private String userIdentifier;
+    private EventType eventType;
+    private long timestamp;
+    private String sender;
```

Comment:
I don't think "sender" and "recipient" are fields that are going to be shared among all event types. I would have EventType and timestamp be the only common fields, and put everything else in the "context" map. Perhaps to make it typesafe, you could have the context map be keyed by an enum that enumerates all known event data fields?